colpali_engine/collators/custom_collator.py

```diff
@@ -85,18 +85,34 @@ def forward_vision_idefics(self, examples):
             text_query = None
             if example["query"] is not None:
                 query = example["query"] + self.suffix
-                messages_query = [
-                    {
-                        "role": "user",
-                        "content": [
-                            {
-                                "type": "text",
-                                "text": f"Question: {query}",
-                            },
-                        ],
-                    },
-                ]
-                text_query = self.processor.apply_chat_template(messages_query, add_generation_prompt=False).strip()
+                if self.processor.__class__.__name__ == "Idefics3Processor":
```

Comment:
if you don't need type image, seems idefics2 and idefic3 are exactly the same so why duplicate ?

Reply:
i think we need type image